pkg/manager/manager.go

```diff
@@ -263,6 +263,10 @@ type Options struct {
 	// +optional
 	Controller config.Controller
 
+	// HookTimeout is the duration given to each hook to return successfully.
+	// To use hooks without timeout, set to a negative duration, e.g. time.Duration(-1)
```

Comment:
>	// To use hooks without timeout, set to a negative duration, e.g. time.Duration(-1)

Can you add a test for this? It looks like the current state of this code would result in a context that is already cancelled if provided with a negative duration.

Reply:
I'll work on a test, but the documented behavior of the context package is that negative timeouts are semantically equivalent to the parent context.